### DISCUSSION TITLE
Is it possible to configure the number of S3 connection threads?

### DISCUSSION THREAD

**Discussion by ggogel** [vmware-tanzu/velero#8175] [Community support Q&A]
Hello,

I've set up velero using the backup storage location provider AWS. As a target, I'm not using an AWS S3 bucket but Backblaze B2. The backup works fine but is very slow. This is due to a limitation of Backblaze B2 allowing only 10Mbit/s per connection thread. This limitation already caused me headaches many times in the past. Looking at the backup progress, it's roughly using 10 Mbit/s, leading me to conclude that velero only uses one thread.

I looked through the documentation for a parameter to configure the number of threads but couldn't find any. Is it possible to configure this?

**Comment by blackpiglet** [comment]
Are you looking for this?
https://velero.io/docs/v1.14/backup-reference/#parallel-files-upload

**Reply by ggogel** [reply]
Thank you for your reply. 

This is not quite what I'm looking for. Multiple threads would also allow to upload multiple parts of a file in parallel. It is described here https://www.backblaze.com/blog/increasing-thread-count-useful-in-sheets-and-cloud-storage-speed/

Longhorn also added this feature recently, see https://github.com/longhorn/longhorn/issues/5189
